pos_pricelist/models/pos_pricelist.py

```diff
@@ -26,3 +31,35 @@ class PosPriceListConfig(models.Model):
         string='Price With Taxes',
         help="Display Prices with taxes on POS"
     )
+
+
+class product_pricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    pos_config_ids = fields.Many2many('pos.config', 'pricelist_posconfig_rel', 'pricelist_id', 'pos_id', string='PoS', help='if empty will be available for all the pos')
+
+    @api.model
+    def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
+        """
+        If request is from pos retrieve only the pricelist applicable.
+            If pricelist has none pos_config_ids assume is available for all the pos
+        """
+        if 'pos' in http.request.httprequest.headers.get('Referer', ''):
```

Comment:
Strange way of dealing with this. Better to change JS part to only load desired pricelists

Reply:
@pedrobaeza thanks for your fast reply, I have done this because I couldn't do any domain to match the m2m to retrieve the pricelists that has the pos selected and those which does not have any pos configured. If you could help me with this I would be glad to correct the code.